### PR TITLE
Backoff retrying failed topology change operation

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

Apply a backoff when retrying failed topology change operation.

## Related issues

related #14959

### CODE REVIEW

File: util/src/main/java/io/camunda/zeebe/util/ExponentialBackoffRetryDelay.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ */
+package io.camunda.zeebe.util;
+
+import java.time.Duration;
+
+/** Retry delay strategy that uses {@link ExponentialBackoff} to calculate the next retry delay. */
+public class ExponentialBackoffRetryDelay implements RetryDelayStrategy {
+
+  private final ExponentialBackoff exponentialBackoff;
+  private long currentDelay;
```
Comment (oleschoenburg) [inline util/src/main/java/io/camunda/zeebe/util/ExponentialBackoffRetryDelay.java:16]: :wrench: Explicitly initialize to `0` here

**Review by lenaschoenburg — APPROVED** [review]
LGTM :+1: 

Not sure about the default values of 10 seconds to 1 minute but I also don't have a better suggestion.

**Comment by deepthidevaki** [comment]
> Not sure about the default values of 10 seconds to 1 minute but I also don't have a better suggestion.

Let's revisit if we see any issues.

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**